### ISSUE TITLE
Add new indicators for logged in users on /podcasts

### ISSUE THREAD

**Issue by kylepolich** [data-skeptic/dataskeptic.com#407]
# Project

We want to promote on-site listening by giving useful features to logged in users connected to dataskeptic.com both on desktop and mobile.  The features will include:

1) Tracking of progress (remember how much of each episode the listener has listened to)
2) Memory of which episodes have been played / never played
3) Ability for user to mark an episode as played
4) Ability for user to mark an episode as a favorite
5) Ability for user to put episodes in a playlist

## Design

![playlist](https://user-images.githubusercontent.com/1054674/35353185-354ab116-00fb-11e8-8a94-93f1b5e4ef3a.png)

Here is the layout we want (images in S3 will be updated later).  Use EXACT links provided.  Designer will replace same file later.  Not that `favorited.png` and `not-favorited.png` are not shown in the design above.  These should be placed to the right of the headphones icon.

Images:

https://s3.amazonaws.com/dataskeptic.com/img/uiux/podcast-list-item/hover.png
https://s3.amazonaws.com/dataskeptic.com/img/uiux/podcast-list-item/played.png
https://s3.amazonaws.com/dataskeptic.com/img/uiux/podcast-list-item/playlist.png
https://s3.amazonaws.com/dataskeptic.com/img/uiux/podcast-list-item/unplayed.png
https://s3.amazonaws.com/dataskeptic.com/img/uiux/podcast-list-item/favorited.png
https://s3.amazonaws.com/dataskeptic.com/img/uiux/podcast-list-item/not-favorited.png

### Get user level data
Call this API when user first logs in and store data in `Cache`.  When updates are made (using API calls), you should update the server AND the local `Cache` at the same time.

GET: `/dev/user/played/list?email=kyle@dataskeptic.com?email=kyle@dataskeptic.com&limit=20&offset=0`

GET: `/api/user/playlist/list?email=kyle@dataskeptic.com`

GET: `/api/user/favorites/list?email=kyle@dataskeptic.com`

Note that `completed` will be either `0.0` or `1.0` guaranteed.  In the future, we might indicate the progress made in the episode, but for now, it is treated as binary for practical purposes but transmitted as float for future planning.

### Mark as played
Call this API when the user makes a chance or completes an episode in the web player.  It is preferred that you send the `blog_id`.  Second preference is to send the `guid`.  If all else fails, you can send the `media` which is a URL to the audio, but DO NOT send this without prior discussion with Kyle about why it is not possible to send either `blog_id` or `guid`.  Only one of these properties should be sent.  The example below incorrectly shows all three for demonstration purposes.

POST: `/api/user/played/update`
REQUEST: `{"email": "kyle@dataskeptic.com", "blog_id": 234, "guid": "not preferred", "media": "http://backup-only-if-blogid-impossible-dont-send-like-this.mp3", "played": 1.0}
RETURNS: `{"status": "ok"}`

This API needs to be called based on the user clicking the button on the Podcast page AND called when the Player reaches the end of an episode.

### Favorite
Call this API when the user clicks the favorite link.

POST: `/api/user/favorite/update`
REQUEST: `{"email": "kyle@dataskeptic.com", "blog_id": 234, "favorited": True}
RETURNS: `{"status": "ok"}`

### Add to Playlist
Call this API when the user clicks the "Add to Playlist" link.

POST: `/api/user/playlist/update`
REQUEST: `{"email": "kyle@dataskeptic.com", "blog_id": 234, "add": True}
RETURNS: `{"status": "ok"}`

## Implementation

Changes need to be made in `/shared/Podcasts` and in `/shared/Player`.  If any changes are needed in the rest of the project, talk to Kyle BEFORE starting any of that work.

### Playlist option

Add a new menu item in the "Membership" menu (which is a dropdown if and only if the user is logged in).  The new menu item is called "Playlist".

If user has no playlist, provide the message "You don't have anything in your playlist.  Please click one of the buttons below"  First button says "Add all episodes" which puts every podcast episode into the playlist.  Second button says "Add all episodes for 20XX" which XX is the current year.  Third button says "Visit podcast page"

If the user has playlist items, display them in order with play buttons.

## Future work (Not in this ticket)

Add to playlist from Player

Create an explainer video showing the interface

**Comment by kylepolich** [comment]
@glozhkovoy - Do not start this project yet.  Please review and ask any questions or provide any feedback.  We want to come up with an estimate for this project.  After the estimate discussion is ready, then we will start the work.

**Comment by kylepolich** [comment]
@glozhkovoy Assigning to you as this is ready to start.  I made subbed API endpoints.  They don't persist, but they should enable you to get started.

**Comment by glozhkovoy** [comment]
@kylepolich I blocked on two steps:

1.) Add all episodes buttons 
I think the best solution is the single endpoint

Request might be similar to ```{"email": "kyle@dataskeptic.com", type: "all" || "18"}```

2.) Playlist episodes

Current playlist is a set of blog_id's. Do we have an endpoint to fetch the episode/blog by id (not prettyname)?

I tried `blog/list?blog_id` but it's seems like not correct solution.
